src/Microsoft.AspNetCore.Server.IISIntegration/Server/IISHttpContext.cs

```diff
@@ -53,14 +56,18 @@ internal abstract partial class HttpProtocol : NativeRequestContext, IDisposable
 
         protected int _requestAborted;
 
+        private CurrentOperationType _currentOperationType;
+        private Task _currentOperation = Task.CompletedTask;
+
         internal unsafe HttpProtocol(PipeFactory pipeFactory, IntPtr pHttpContext)
-            : base((HttpApiTypes.HTTP_REQUEST*) NativeMethods.http_get_raw_request(pHttpContext))
+            : base((HttpApiTypes.HTTP_REQUEST*)NativeMethods.http_get_raw_request(pHttpContext))
         {
             _thisHandle = GCHandle.Alloc(this);
 
             _pipeFactory = pipeFactory;
             _pHttpContext = pHttpContext;
 
+            NativeMethods.http_set_managed_context(pHttpContext, (IntPtr)_thisHandle);
```

Comment:
Creates the stored context in native. Native will use it for on async completion (and eventually abort).